nix/haskell-overlay.nix

```diff
@@ -0,0 +1,8 @@
+{ sources ? import ./sources.nix, pkgs }:
```

Comment:
Minor nit: Since none of the haskell dependencies for `dbcritic` are things we pin with `niv`, you don't need to include `sources` as an argument here (since we won't use it)